docs: Refactor-homepage-architecture

rockument69 opened the pull request:
Home page does not adequately describe the reasons for a developer to use Angular,
and there are no code samples.
Issue (GitHub): [31774](https://github.com/angular/angular/issues/31774)
Issue (Jira): DOCS-696
The home page now includes new key points to describe reasons for using Angular,
and a highlighted code sample.

## PR Checklist
Please check if your PR fulfills the following requirements:

- [x] The commit message follows our guidelines: https://github.com/angular/angular/blob/master/CONTRIBUTING.md#commit
- [ ] Tests for the changes have been added (for bug fixes / features)
- [ ] Docs have been added / updated (for bug fixes / features)


## PR Type
What kind of change does this PR introduce?

<!-- Please check the one that applies to this PR using "x". -->

- [ ] Bugfix
- [ ] Feature
- [ ] Code style update (formatting, local variables)
- [ ] Refactoring (no functional changes, no api changes)
- [ ] Build related changes
- [ ] CI related changes
- [x] Documentation content changes
- [ ] angular.io application / infrastructure changes
- [ ] Other... Please describe:


## What is the current behavior?
<!-- Please describe the current behavior that you are modifying, or link to a relevant issue. -->

Issue Number: [DOCS-696](https://angular-team.atlassian.net/browse/DOCS-696)

Home page does not adequately describe the reasons for a developer to use Angular, and there are no code samples.

## What is the new behavior?

The home page now includes new key points to describe reasons for using Angular, and a highlighted code sample. The architecture page now includes a new introduction and diagram for beginners.

For details see [Doc Plan: Refresh home page](https://docs.google.com/document/d/1cqyLl59BFyjnpUZD3NxkyBB4CjJrt7XXg9c6uhRq4HA/edit?usp=sharing).

## Does this PR introduce a breaking change?

- [ ] Yes
- [x] No


<!-- If this PR contains a breaking change, please describe the impact and migration path for existing applications below. -->


## Other information

rockument69 committed:
docs: Refactor-homepage-architecture

Home page does not adequately describe the reasons for a developer to use Angular,
and there are no code samples.
Issue Number: DOCS-696
The home page now includes new key points to describe reasons for using Angular,
and a highlighted code sample.

mary-poppins commented:
You can preview 4cb6cb0 at https://pr36823-4cb6cb0.ngbuilds.io/.

jbogarthyde commented:
I think it looks strange to have the stackblitz app embedded, then have the link go to the walkthrough with no explanation -- maybe put in some more context, something like:

 "Here's a small sample app in StackBlitz that shows how your data model is directly connected to a dynamic page display."
       <embedded sample>
"If you want to explore further without setting up a full development environment, try this walk-through."
    <Try it now>
This mini-tutorial lets you load this sample into StackBlitz, and use it to experience how you work with an Angular app. 
In addition to getting an intro to the basics, you can walk through adding features such in-app view navigation and interaction with a server."

mary-poppins commented:
You can preview f52c0c0 at https://pr36823-f52c0c0.ngbuilds.io/.

rockument69 commented:
> I think it looks strange to have the stackblitz app embedded, then have the link go right to the walk-through with no explanation of the relationship. Maybe put in some more context, something like:
> 
> Here's a small sample app in StackBlitz that shows how your data model is directly connected to a dynamic page display.
> 
> ```
>  **embedded sample**
> ```
> 
> If you want to explore further without setting up a full development environment, try this walk-through.
> 
> ```
>   **Try it now**
> ```
> 
> This mini-tutorial lets you load this sample into StackBlitz, and use it to experience how you work with an Angular app.
> In addition to getting an intro to the basics, you can walk through adding features such in-app view navigation and interaction with a server."

OK, see new version. Thanks!

mary-poppins commented:
You can preview 2f5dce6 at https://pr36823-2f5dce6.ngbuilds.io/.

jbogarthyde commented:
Ok, much simpler, but totally works 👍

gkalpak commented on aio/content/marketing/index.html:
```diff
@@ -100,14 +101,33 @@ <h1 class="no-anchor no-toc" style="display: none"></h1>
+    </div>
+    
+    <!-- Group 6 (embeds the StackBlitz live example) -->
+    <div class="text-container">
```
Why have this outside ` .home-row`?

rockument69 replied:
The embedded example renders narrow if I designate it as a `row` or `.home-row`. I changed it to `layout="wide"` so that it renders correctly.

gkalpak commented on aio/content/marketing/index.html:
```diff
@@ -100,14 +101,33 @@ <h1 class="no-anchor no-toc" style="display: none"></h1>
+    <!-- Group 5 (introduces the live example) -->
+    <div layout="row" class="home-row">
+      <div class="text-container">
+          <div class="text-headline">Try it now.</div>
+          <p class="text-body">Explore Angular's capabilities with a sample app that shows how your data model can directly connect to a dynamic page display.
+            No setup required.
+          </p>
```
Inconsistent indentation.

rockument69 replied:
OK thanks

gkalpak commented on aio/content/marketing/index.html:
```diff
@@ -100,14 +101,33 @@ <h1 class="no-anchor no-toc" style="display: none"></h1>
+          </p>
+        </div>
+      </div>
+    </div>
```
One too extra `</div>`

rockument69 replied:
Ok fixed.

gkalpak commented on aio/content/marketing/index.html:
```diff
@@ -100,14 +101,33 @@ <h1 class="no-anchor no-toc" style="display: none"></h1>
+    <div layout="row" class="home-row">
+      <div class="text-container">
+          <div class="text-headline">Try it now.</div>
+          <p class="text-body">Explore Angular's capabilities with a sample app that shows how your data model can directly connect to a dynamic page display.
```
```suggestion
          <p class="text-body">
            Explore Angular's capabilities with a sample app that shows how your data model can directly connect to a dynamic page display.
```

gkalpak reviewed (commented):
FWIW, the `angular-rough-diagram.png` seems confusing to me 😇

mary-poppins commented:
You can preview fec6803 at https://pr36823-fec6803.ngbuilds.io/.

rockument69 commented:
> Ok, much simpler, but totally works 👍
> Typo in last part -- features such **as** ...

OK fixed.

mary-poppins commented:
You can preview 1226cb1 at https://pr36823-1226cb1.ngbuilds.io/.

mary-poppins commented:
You can preview 29168d7 at https://pr36823-29168d7.ngbuilds.io/.

gkalpak commented on aio/content/marketing/index.html:
```diff
@@ -100,14 +101,41 @@ <h1 class="no-anchor no-toc" style="display: none"></h1>
+    <!-- Group 5 (introduces the live example) -->
+    <div layout="row" class="home-row">
+      <div class="text-container">
+<<<<<<< HEAD
+          <div class="text-headline">Try it now.</div>
+          <p class="text-body">
+            Explore Angular's capabilities with a sample app that shows how your data model can directly connect to a dynamic page display.
+            No setup required.
+=======
+          <div class="text-headline">Try the sample app.</div>
+          <p class="text-body">
+            Explore Angular's capabilities with a sample app that shows how your data model can directly connect to a dynamic page display. No setup required.
+>>>>>>> fixup! docs: Refactor-homepage-architecture
```
Rebase conflict markers :grin:

rockument69 replied:
OK, squashed commits and fixed this conflict.

gkalpak commented on aio/content/marketing/index.html:
```diff
@@ -100,14 +101,41 @@ <h1 class="no-anchor no-toc" style="display: none"></h1>
+
+    <!-- Group 6 (the live example should be wide, not a row) -->
+    <div layout="wide">
+        <live-example name="getting-started" embedded></live-example>
```
Inconsistent indentation.

rockument69 replied:
OK fixed

mary-poppins commented:
You can preview f70c726 at https://pr36823-f70c726.ngbuilds.io/.

rockument69 commented:
> FWIW, the `angular-rough-diagram.png` seems confusing to me 😇

Looking back from a rendered web page, I arranged the HTML/CSS, Components, and Services to show fundamental relationships. The diagram is meant to accompany the text that introduces these elements, and it is done in the style of other diagrams (the one at the bottom of this page, and other diagrams on other Fundamentals pages). It is meant only to be a high-level depiction, not going into depth as the other diagrams already do.

mary-poppins commented:
You can preview c17ec3e at https://pr36823-c17ec3e.ngbuilds.io/.

mary-poppins commented:
You can preview 7d9eb95 at https://pr36823-7d9eb95.ngbuilds.io/.

gkalpak commented on aio/content/marketing/index.html:
```diff
@@ -100,14 +101,34 @@ <h1 class="no-anchor no-toc" style="display: none"></h1>
+
+    <!-- Group 6 (the live example should be wide, not a row) -->
+    <div layout="wide">
+      <live-example name="getting-started" embedded></live-example>
```
It turns out that this brings in too much code, which affects the performance of the homepage so severely (esp. on slower networks) that the user experience cannot be considered reasonable for a PWA (hence the failing PWA score tests).

You can download [this report][1] and load it into the [Lighthouse viewer][2] to see more details. Notice that on slower networks it can take more than 20 seconds for the page to load.

Note that the homepage is by far the most visited page (based or Google analytics), so we can assume it is visited by people that are already know about Angular. Therefore, it is important to keep it lean and fast to load.

---
So, here are the options I can think of (off the top of my head):

## 1. Non-embedded example
Make the example non-embedded (i.e. remove the `embedded` attribute from the `<live-example>` element). This avoids loading the bulk of the extra code upfront (making the PWA score tests pass again), but it still requires loading a custom element (the `<live-example>` element), which adds some overhead (even if minimal).

**Pros:**
- Example available from the homepage.

**Cons:**
- Not embedded (the link opens in a new tab).
- Loads some extra code on the homepage, which most users won't use.

## 2. New embedded example component
If you consider it crucial to have the embedded example on the homepage, we could look into creating a component that can show a placeholder image without loading the extra code for running the example, until the user clicks on the example. (This is a similar experience with the current embedded example, except we would have to implement it ourselves instead of relying on StackBlitz's embedding capabilities).

**Pros:**
- Example embedded into the homepage.

**Cons:**
- Extra implementation/maintenance cost.
- Loads some extra code on the homepage, which most users won't use.

## 3. Hard-code link to live example
Instead of embedding the example into the homepage, we could dd links either to the `Getting started` guide or to the live example directly (i.e. with links to the appropriate [stackblitz.html][3] and [getting-started.zip][4] files). This would basically offer the same experience as option 1, but without having to load the `<live-example>` custom element - i.e. no extra code.

**Pros:**
- Example available from the homepage.
- No extra code that users might not need anyway.

**Cons:**
- Not embedded (the link opens in a new tab).

[1]: https://github.com/angular/angular/files/4558885/angular.io-results.json.txt
[2]: https://googlechrome.github.io/lighthouse/viewer
[3]: https://angular.io/generated/live-examples/getting-started/stackblitz.html
[4]: https://angular.io/generated/zips/getting-started/getting-started.zip

mary-poppins commented:
You can preview 361074e at https://pr36823-361074e.ngbuilds.io/.

mary-poppins commented:
You can preview 50651cf at https://pr36823-50651cf.ngbuilds.io/.

mary-poppins commented:
You can preview 0b7e608 at https://pr36823-0b7e608.ngbuilds.io/.

mary-poppins commented:
You can preview 8ff0ae6 at https://pr36823-8ff0ae6.ngbuilds.io/.

mary-poppins commented:
You can preview 2c6ea73 at https://pr36823-2c6ea73.ngbuilds.io/.

mary-poppins commented:
You can preview 30617e0 at https://pr36823-30617e0.ngbuilds.io/.

mary-poppins commented:
You can preview 1e3441a at https://pr36823-1e3441a.ngbuilds.io/.

mary-poppins commented:
You can preview 4fff3f0 at https://pr36823-4fff3f0.ngbuilds.io/.

mary-poppins commented:
You can preview 373295b at https://pr36823-373295b.ngbuilds.io/.

mary-poppins commented:
You can preview 08ba2b0 at https://pr36823-08ba2b0.ngbuilds.io/.

jelbourn commented:
Just doing a drive-by, I was originally the one who filed the Jira issue. 

My my problems with the current homepage were:
* Nothing explains what Angular **is** (e.g. "A framework for building web applications")
* There's a heavy emphasis on native mobile, which is not super representative of the state of Angular today.

mary-poppins commented:
You can preview a4a6138 at https://pr36823-a4a6138.ngbuilds.io/.

rockument69 commented on aio/content/marketing/index.html:
```diff
@@ -100,14 +101,34 @@ <h1 class="no-anchor no-toc" style="display: none"></h1>
+          <div class="text-headline">Try the sample app.</div>
+          <p class="text-body">
+            Explore Angular's capabilities with a sample app that shows how your data model can directly connect to a dynamic page display. No setup required.
+          </p>
```
OK done.

mary-poppins commented:
You can preview 26951ca at https://pr36823-26951ca.ngbuilds.io/.

rockument69 commented:
> Just doing a drive-by, I was originally the one who filed the Jira issue.
> 
> My my problems with the current homepage were:
> 
> * Nothing explains what Angular **is** (e.g. "A framework for building web applications")
> * There's a heavy emphasis on native mobile, which is not super representative of the state of Angular today.

Thanks for your comment. I added "Angular is a framework for building single-page apps." It's the first sentence under "Develop Across All Platforms". I also removed "native mobile" from the list, so it is now "Run the apps on web, mobile web, and native desktop platforms."

jelbourn commented:
Native desktop falls into the same bucket as native mobile. I think @StephenFluin and/or @mgechev should chime in on exact wording

mary-poppins commented:
You can preview 9aaed19 at https://pr36823-9aaed19.ngbuilds.io/.

mgechev commented on aio/content/guide/architecture.md:
```diff
@@ -1,75 +1,101 @@
+  The code examples in this topic are available as a <live-example></live-example>.
+</div>
+
+Angular is an [open source](https://en.wikipedia.org/wiki/Open-source "Open source") cross-platform [framework](https://en.wikipedia.org/wiki/Web_framework "Web framework") for building single-page applications using [HTML](https://developer.mozilla.org/docs/Learn/HTML/Introduction_to_HTML "Introduction to HTML") with [JavaScript](https://developer.mozilla.org/en-US/docs/Web/JavaScript/A_re-intro ...
```
I'd focus on the web aspect instead of presenting Angular as a cross-platform. @IgorMinar should have an opinion here as well.

IgorMinar replied:
Exactly. Angular is a solution for building UIs for the Web. And the Web is a cross-platform medium.

rockument69 replied:
OK, rewrote the intro paragraph as follows:

> Angular is an [open source](https://en.wikipedia.org/wiki/Open-source "Open source") [framework](https://en.wikipedia.org/wiki/Web_framework "Web framework") and a platform of components and tooling for building web applications using [HTML](https://developer.mozilla.org/docs/Learn/HTML/Introduction_to_HTML "Introduction to HTML") with [JavaScript](https://developer.mozilla.org/en-US/docs/Web/JavaScript/A_re-introduction_to_JavaScript "A re-introduction to JavaScript") or [TypeScript](https://www.typescriptlang.org/ "TypeScript: JavaScript that scales").
> Its architecture enables the apps to run across different environments on clients and servers using a JavaScript engine.
> Angular is supported by the Angular Team at Google and by a community of individuals and corporations.
>

mgechev commented on aio/content/guide/architecture.md:
```diff
@@ -1,75 +1,101 @@
+</div>
+
+Angular is an [open source](https://en.wikipedia.org/wiki/Open-source "Open source") cross-platform [framework](https://en.wikipedia.org/wiki/Web_framework "Web framework") for building single-page applications using [HTML](https://developer.mozilla.org/docs/Learn/HTML/Introduction_to_HTML "Introduction to HTML") with [JavaScript](https://developer.mozilla.org/en-US/docs/Web/JavaScript/A_re-intro ...
+It's architecture enables the apps to run across different environments—browser, server, web-worker, and mobile devices—using a JavaScript engine.
```
WebWorker is an experimental use case, mobile is supported, but mostly focus of third-parties. I'd be a little more conservative - "run on the client and the server".

IgorMinar replied:
correct, we deprecated the webworker support as our current the solution is suboptimal. This doesn't mean that we won't come back to webworkers in the future, we might, but for now it's not something that is mainstream and something we want to encourage people to use.

rockument69 replied:
Fixed in the above rewritten intro:

> Its architecture enables the apps to run across different environments on clients and servers using a JavaScript engine.

mgechev commented on aio/content/guide/architecture.md:
```diff
@@ -1,75 +1,101 @@
+An app always has at least a *root* module that enables bootstrapping, and typically has many more *feature modules*.
 
-* Components define *views*, which are sets of screen elements that Angular can choose among and modify according to your program logic and data.
+An NgModule provides a compilation context for [*components*](#components), which define views as sets of screen elements that Angular can choose among and modify according to your program logic and data.
```
I tend to think of NgModules more as the lexical scope of the templates of the declared components inside of them. @alxhub could correct me if I'm wrong.

IgorMinar replied:
both @rockument69's and your proposal are technically correct. I think Tony's explanation is simpler to understand.

rockument69 replied:
I actually shortened it, because the 2nd sentence about the root module is repeated below, in the section about NgModules. Therefore, the paragraph is now as follows:

> You define an Angular app using a set of [*NgModules*](#modules), which collect related code into functional sets.
> An NgModule provides a compilation context for [*components*](#components), which define views as sets of screen elements that Angular can choose among and modify according to your program logic and data.

Links take the reader to the actual definitions of these terms.

pkozlowski-opensource replied:
Hmm, I would probably start with _components_. In my head everything starts with a component. Components can use services. And modules group components + services. I find it a bit odd to start with modules since those are currently architectural / impl _necessity_ rather than the central concept.

IgorMinar commented on aio/content/guide/architecture.md:
```diff
@@ -1,75 +1,101 @@
+  The code examples in this topic are available as a <live-example></live-example>.
+</div>
+
+Angular is an [open source](https://en.wikipedia.org/wiki/Open-source "Open source") cross-platform [framework](https://en.wikipedia.org/wiki/Web_framework "Web framework") for building single-page applications using [HTML](https://developer.mozilla.org/docs/Learn/HTML/Introduction_to_HTML "Introduction to HTML") with [JavaScript](https://developer.mozilla.org/en-US/docs/Web/JavaScript/A_re-intro ...
```
Angular is much more than a framework at this point.

Angular = framework + components + tooling + community.

rockument69 replied:
Added in the above paragraph. "...an open-source framework and a platform of components and tooling for..." The last sentence describes "... supported ... by a community..."

IgorMinar commented on aio/content/guide/architecture.md:
```diff
@@ -1,75 +1,101 @@
 
-Before a view is displayed, Angular evaluates the directives and resolves the binding syntax in the template to modify the HTML elements and the DOM, according to your program data and logic. Angular supports *two-way data binding*, meaning that changes in the DOM, such as user choices, are also reflected in your program data.
+Before a view is displayed, Angular evaluates the directives and resolves the binding syntax in the template to modify the HTML elements and the DOM, according to your program data and logic.
+Angular supports *two-way data binding*, meaning that changes in the DOM, such as user choices, are also reflected in your program data.
```
```suggestion
Angular supports *data binding*, meaning updated to the application state are automatically reflected in the view via DOM updates. Angular also supports *event binding* which allows a user input to update the application state.
```

IgorMinar replied:
the two way data binding topic is slightly controversial because a/ Angular doesn't support "true" two-way data-binding, we provide a feature that is similar to it, and b/ two-way data binding is now seen as an anti-pattern and not a feature.

rockument69 replied:
OK. I rewrote so that I would not be adding data and event binding twice:

> A [template](/guide/glossary#template "Definition of template") combines HTML with Angular markup that can modify HTML elements before they are displayed.
> Template *directives* provide program logic, and *binding markup* connects your app's data and the DOM.
> There are two types of data binding:
> 
> * *Event binding* lets your app respond to user input in the target environment by updating your application data.
> * *Property binding* lets you interpolate values that are computed from your application data into the HTML.
> 
> Before a view is displayed, Angular evaluates the directives and resolves the binding syntax in the template to modify the HTML elements and the DOM, according to your program data and logic.
> Updates to the app state are automatically reflected in the view through DOM updates.
>

IgorMinar reviewed (dismissed):
This looks like a good cleanup! Thanks Tony!

I left a few comments. Additionally, I agree with @gkalpak that the diagram is confusing. What goal are you aiming to achieve with the diagram?

mary-poppins commented:
You can preview 109f873 at https://pr36823-109f873.ngbuilds.io/.

rockument69 commented:
> This looks like a good cleanup! Thanks Tony!
> 
> I left a few comments. Additionally, I agree with @gkalpak that the diagram is confusing. What goal are you aiming to achieve with the diagram?

The current doc uses a diagram (Figure 2 in my draft) that I didn't understand. In Figure 1 I try to bridge the gap from no diagram to the Figure 2 diagram, using the same drawing style.

I'm a complete novice, which is more like the audience for this page -- people who may have some background but are completely new to Angular. In Figure 1 I am trying to show the relationships between a service, a component, a template, and the result (a web page or portion of one). The diagram appears next to the simple paragraphs that introduce these terms. The goal is to introduce a complete novice to what Angular does. Basically you use a service with a data source, a component for the code, and a template for physical layout of the web page. 

If we can somehow modify Figure 1, we can keep it. If this diagram doesn't work, or we don't want such a diagram, I think we should also remove Figure 2, which isn't useful unless there is a Figure 1 that leads the reader to it.

rockument69 commented:
> Native desktop falls into the same bucket as native mobile. I think @StephenFluin and/or @mgechev should chime in on exact wording

I will remove "native" and use the following sentence:

> Angular is a framework with components and tooling for building web applications that run across mobile and desktop platforms

mary-poppins commented:
You can preview f3df004 at https://pr36823-f3df004.ngbuilds.io/.

jelbourn commented on aio/content/marketing/index.html:
```diff
@@ -53,9 +53,10 @@ <h1 class="no-anchor no-toc" style="display: none"></h1>
-          <p class="text-body">Angular puts you in control over scalability. Meet huge data requirements by building data models on RxJS, Immutable.js or another push-model.</p>
+          <div class="text-headline">Scale up performance with data.</div>
+          <p class="text-body">Achieve maximum speed and functionality by using data models, data binding, logical directives, web workers, and server-side rendering.
+            For more details, see <a href="features">Features</a>.
```
I feel like this block is mostly framework jargon that doesn't help explain what the framework actually offers. I would replace this block with something that talks about Angular's expanded ecosystem features.

rockument69 replied:
OK:
- Using the new tagline, thanks!

- First block:
Use an open source framework with a vibrant community.
Trusted by over 2000 projects at Google, Angular is is a safe bet for your next project.

- Second block:
Build and deploy apps quickly.
Use HTML declarative templates with CSS styles to quickly create UI views, and TypeScript components to generate code highly optimized for JavaScript virtual machines. For an overview, see the <a href="guide/architecture#introduction-to-angular-concepts">Introduction to Angular concepts</a>.

- Third block:
Scale performance with rich features.
Deploy the router to map paths to views, forms to handle user input, animation to enhance your app's UX, and client services for streamlined error handling and testing. For more details, see <a href="features">Features</a>.

- Fourth block:
Benefit from the combined talents of the vast Angular ecosystem.
Choose from a wide variety of existing components and learn about best practices. Get immediate Angular-specific help and feedback with nearly every IDE and editor.

jelbourn commented on aio/content/marketing/index.html:
```diff
@@ -75,8 +76,9 @@ <h1 class="no-anchor no-toc" style="display: none"></h1>
-          <div class="text-headline">Incredible Tooling</div>
-          <p class="text-body">Build features quickly with simple, declarative templates. Extend the template language with your own components and use a wide array of existing components. Get immediate Angular-specific help and feedback with nearly every IDE and editor. All this comes together so you can focus on building amazing apps rather than trying to make the code work.
+          <div class="text-headline">Build and deploy apps quickly.</div>
+          <p class="text-body">Use HTML declarative templates to quickly create UI views, and JavaScript/TypeScript components to generate code highly optimized for JavaScript virtual machines.
```
At the very least I would change "JavaScript/TypeScript" to just TypeScript here. Going further, I don't think that VM Optimization is really a homepage-worthy selling point.

rockument69 replied:
OK, edited (see Second block, above comment).

jelbourn commented on aio/content/marketing/index.html:
```diff
@@ -100,14 +104,40 @@ <h1 class="no-anchor no-toc" style="display: none"></h1>
+    <div layout="row" layout-xs="column" class="home-row">
+      <div class="promo-img-container promo-3">
+        <div><a href="generated/live-examples/getting-started/stackblitz.html" target="_blank" title="Live sample app">
+          <img src="generated/images/marketing/home/getting-started-code.png" alt="Image of the live sample app"></a>
```
This image doesn't really look great (the screenshot looks distorted).

rockument69 replied:
OK, tried creating a better screen shot.

jelbourn reviewed (commented):
I don't know if @StephenFluin / @mgechev  / @IgorMinar will agree, but I personally still find that tagline to be a bit cumbersome, since specifying "web" feels like enough. I also think we should replace the hero tagline ("One framework. Mobile & desktop.") with something much more descriptive. 

My pitch would be:
* **Tagline**: A framework for building web applications
**First sub-block**: "Open source with a vibrant community". Potentially mention who uses Angular.
**Second sub-block**: "Build and deploy apps quickly." Explicitly emphasize developer experience here, as well as using technologies people know and love (HTML, CSS, TS).
**Third sub-block**: "Rich features" - talking about forms, routing, animations, HTTP client.

otodockal commented:
Can we add "2000+ Angular Projects @ Google" from Jules ng-conf presentation? :-) 
People need to know Angular is a safe bet for their next projects. Google's adoption can help.
Only idea :-).

or as Jeremy mentioned, who uses Angular, I think it's a good idea. https://www.typescriptlang.org/ is also a good inspiration.

mgechev commented:
I agree with all suggestions of @jelbourn.

>First sub-block: "Open source with a vibrant community". Potentially mention who uses Angular.

I'd also add what @otodockal mentioned:

>Trusted by 2000+ Projects at Google

mary-poppins commented:
You can preview d94d55c at https://pr36823-d94d55c.ngbuilds.io/.

mary-poppins commented:
You can preview 72a0811 at https://pr36823-72a0811.ngbuilds.io/.

mary-poppins commented:
You can preview 5913c2b at https://pr36823-5913c2b.ngbuilds.io/.

mary-poppins commented:
You can preview daeab42 at https://pr36823-daeab42.ngbuilds.io/.

mary-poppins commented:
You can preview 9a7e89b at https://pr36823-9a7e89b.ngbuilds.io/.

mary-poppins commented:
You can preview fe250ff at https://pr36823-fe250ff.ngbuilds.io/.

mary-poppins commented:
You can preview 4b8e78e at https://pr36823-4b8e78e.ngbuilds.io/.

mary-poppins commented:
You can preview 8a8c4d1 at https://pr36823-8a8c4d1.ngbuilds.io/.

mary-poppins commented:
You can preview b4e9df9 at https://pr36823-b4e9df9.ngbuilds.io/.

mary-poppins commented:
You can preview ac3ec79 at https://pr36823-ac3ec79.ngbuilds.io/.

mary-poppins commented:
You can preview 4626efd at https://pr36823-4626efd.ngbuilds.io/.

rockument69 commented:
> This looks like a good cleanup! Thanks Tony!
> 
> I left a few comments. Additionally, I agree with @gkalpak that the diagram is confusing. What goal are you aiming to achieve with the diagram?

@IgorMinar : Thanks for your feedback! I edited the diagram to show the conceptual parts of an NgModule with services, removing the rest to make it simple. The design is similar to the existing Figure 2 at the bottom of the page.

Please review and let me know if this works for you (and for a new reader who has no idea what Angular is). Thanks!

mary-poppins commented:
You can preview a98139d at https://pr36823-a98139d.ngbuilds.io/.

mary-poppins commented:
You can preview 292fcda at https://pr36823-292fcda.ngbuilds.io/.

pkozlowski-opensource commented on aio/content/guide/architecture.md:
```diff
@@ -1,75 +1,101 @@
+</div>
 
-* Components define *views*, which are sets of screen elements that Angular can choose among and modify according to your program logic and data.
+Angular is an [open source](https://en.wikipedia.org/wiki/Open-source "Open source") [framework](https://en.wikipedia.org/wiki/Web_framework "Web framework") and a platform of components and tooling for building web applications using [HTML](https://developer.mozilla.org/docs/Learn/HTML/Introduction_to_HTML "Introduction to HTML") with [JavaScript](https://developer.mozilla.org/en-US/docs/Web/Jav ...
```
Do you want to call out JavaScript here? Angular apps pretty much require usage of TypeScript so I find it a bit misleading that we list JS and TS on par here.

rockument69 replied:
I agree, changing to "HTML and TypeScript".

pkozlowski-opensource commented on aio/content/guide/architecture.md:
```diff
@@ -1,75 +1,101 @@
 
-* Components use *services*, which provide specific functionality not directly related to views. Service providers can be *injected* into components as *dependencies*, making your code modular, reusable, and efficient.
+Written in TypeScript, Angular implements core and optional functionality as a set of TypeScript libraries that you import into your apps.
+Angular combines declarative templates, dependency injection, end-to-end tooling, and integrated best practices to solve development challenges.
```
nit: "solve" - > "address" / "tackle"? Can one "solve challenges"? (not a native EN speaker so might be nit-picking for nothing here...)

rockument69 replied:
I like "tackle" so using that, thanks!

pkozlowski-opensource commented on aio/content/guide/architecture.md:
```diff
@@ -1,75 +1,101 @@
+* The metadata for a service class provides the information Angular needs to make it available to components through *dependency injection (DI)*.
 
-  See the [Angular Glossary](guide/glossary) for basic definitions of important Angular terms and usage.
+Figure 1 shows a conceptual diagram of a feature NgModule that includes a component, a template, and a service injected into the component.
```
"diagram of a feature NgModule that includes a component, a template" - this sounds like NgModule contains components _and_ templates but it is rather a component that is an "owner" of a template. How about "includes a component _with_ a template"

rockument69 replied:
For the above comments: 

New sentence for introducing components and NgModule:
> You define an Angular app using a set of [*components*](#components), which define views as sets of screen elements that Angular can choose among and modify according to your program logic and data. An [*NgModule*](#modules) groups components into functional sets and provides a compilation context for them. 

Using "with a template" as suggested above.

pkozlowski-opensource commented on aio/content/guide/architecture.md:
```diff
@@ -1,75 +1,101 @@
-Angular *NgModules* differ from and complement JavaScript (ES2015) modules. An NgModule declares a compilation context for a set of components that is dedicated to an application domain, a workflow, or a closely related set of capabilities. An NgModule can associate its components with related code, such as services, to form functional units.
+Angular [NgModules](/guide/glossary#ngmodule "Definition of NgModule") differ from and complement JavaScript (ES2015) modules.
+An NgModule declares a compilation context for a set of components that is dedicated to an application domain, a workflow, or a closely related set of capabilities.
+An NgModule can associate its components with related code, such as services, to form functional units.
```
"An NgModule can associate its components with related code, such as services, to form functional units."

Hmm, this doesn't sound quite right to me... It is not like an NgModule is associating services with components. NgModule makes services _available_ to components, but it is up to a component to use a given service (and thus make an association). Moreover a component can use a service defined in a _different_ NgModule (obvious in in case of lazy-loaded  modules) so yeh, we need to review this part.

rockument69 replied:
Removing the sentence "An NgModule can associate its components with related code, such as services, to form functional units."

pkozlowski-opensource commented on aio/content/guide/architecture.md:
```diff
@@ -1,75 +1,101 @@
 
-Every Angular application has at least one component, the *root component* that connects a component hierarchy with the page document object model (DOM). Each component defines a class that contains application data and logic, and is associated with an HTML *template* that defines a view to be displayed in a target environment.
+Every Angular application has at least one [component](/guide/glossary#component "Definition of component"), the *root component* that connects a component hierarchy with the page document object model (DOM).
+Each component defines a class that contains application data and logic, and is associated with an HTML *template* that defines a view to be displayed in a target environment.
```
I would drop HTML from here - the way it is written sounds like Angular templates contain only HTML.

rockument69 replied:
Done

pkozlowski-opensource commented on aio/content/guide/architecture.md:
```diff
@@ -1,75 +1,101 @@
 * *Property binding* lets you interpolate values that are computed from your application data into the HTML.
 
-Before a view is displayed, Angular evaluates the directives and resolves the binding syntax in the template to modify the HTML elements and the DOM, according to your program data and logic. Angular supports *two-way data binding*, meaning that changes in the DOM, such as user choices, are also reflected in your program data.
+Before a view is displayed, Angular evaluates the directives and resolves the binding syntax in the template to modify the HTML elements and the DOM, according to your program data and logic.
```
"HTML elements and the DOM" <- in this context "HTML elements" and "DOM" are the same thing so we should have only one of those here.

rockument69 replied:
Using DOM. Thanks!

pkozlowski-opensource reviewed (commented):
Left some comments.

mary-poppins commented:
You can preview e471e56 at https://pr36823-e471e56.ngbuilds.io/.

mary-poppins commented:
You can preview d40ae29 at https://pr36823-d40ae29.ngbuilds.io/.

mary-poppins commented:
You can preview 1fd9298 at https://pr36823-1fd9298.ngbuilds.io/.

mary-poppins commented:
You can preview 2e3ec9c at https://pr36823-2e3ec9c.ngbuilds.io/.

pkozlowski-opensource commented on aio/content/guide/architecture.md:
```diff
@@ -1,75 +1,107 @@
 There are two types of data binding:
 
 * *Event binding* lets your app respond to user input in the target environment by updating your application data.
 * *Property binding* lets you interpolate values that are computed from your application data into the HTML.
```
I know that this is the existing text (and not sth modified by this PR) but the "interpolate values" phrasing doesn't "click" with me. How about "lets you map values from your application data into the HTML"

rockument69 replied:
OK, done.

pkozlowski-opensource commented on aio/content/guide/architecture.md:
```diff
@@ -83,12 +115,14 @@ Angular provides predefined pipes for common transformations, and you can also d
+The decorator provides the metadata that allows other providers to be **injected** as dependencies into your class.
 
- *Dependency injection* (DI) lets you keep your component classes lean and efficient. They don't fetch data from the server, validate user input, or log directly to the console; they delegate such tasks to services.
+ *Dependency injection* (DI) lets you keep your component classes lean and efficient.
```
nit: "lean and efficient" sounds a bit like a "marketing slogan" without real meaning. What are we trying to say here? I guess the fact that code can be moved out of the component class (the lean part?). But moving code around doesn't make it more "efficient". I guess the real benefit is that code is split into smaller, cohesive chunks.

Can you try to re-phrase it (again, sorry for the nit-picking, I know that this is pre-existing text)

rockument69 replied:
As for pre-existing text... I started working on the home page and realized that I needed some kind of intro paragraph for the architecture.md page. Reviews are catching all these problems in the existing text. (I'm a tech writer, I should have known better.) This experience is like the following (choose one):
- Opening a can of worms. To re-can, you need a larger can.
- Opening Pandora's box. You have no idea what you might find.
- The remaining work to finish in order to reach your goal increases as the deadline approaches.

Thanks!

pkozlowski-opensource reviewed (approved):
I've added few more comments but it LGTM overall. I will let others to chime in and iterate on this text.
Adding the cleanup label as the CI is failing and I would love to see my comments acknowledged / addressed.

Reviewed-for: public-api
Reviewed-for: fw-core

mary-poppins commented:
You can preview 397781d at https://pr36823-397781d.ngbuilds.io/.

rockument69 commented on aio/content/guide/architecture.md:
```diff
@@ -1,75 +1,107 @@
 
-Modules, components and services are classes that use *decorators*. These decorators mark their type and provide metadata that tells Angular how to use them.
+Building web applications is typically an exercise in tricking the browser to dynamically load portions of the user interface (UI).
+With Angular you can use the [single-page application](https://en.wikipedia.org/wiki/Single-page_application "Single-page application in Wikipedia") (SPA) design pattern to dynamically rewrite the current web page with new portions of the page from the server, which is significantly faster than loading entire pages from the server.
```
Agreed, done in the next commit.

rockument69 commented on aio/content/guide/architecture.md:
```diff
@@ -1,75 +1,107 @@
 
-* The metadata for a component class associates it with a *template* that defines a view. A template combines ordinary HTML with Angular *directives* and *binding markup* that allow Angular to modify the HTML before rendering it for display.
+HTML is a great declarative language for static documents, but it doesn't provide enough functionality to create web applications.
+Angular extends HTML to provide features that eliminate much of the code you would otherwise have to write, such as data binding, dependency injection, form validation, routing, and document object model (DOM) control structures.
```
I'm changing "control structures" to "manipulation", as in "DOM manipulation". This also fits better with the other phrases.

rockument69 commented on aio/content/guide/architecture.md:
```diff
@@ -1,75 +1,107 @@
+Angular extends HTML to provide features that eliminate much of the code you would otherwise have to write, such as data binding, dependency injection, form validation, routing, and document object model (DOM) control structures.
 
-* The metadata for a service class provides the information Angular needs to make it available to components through *dependency injection (DI)*.
+By declaratively describing how your app's UI should change as your app state changes, you are freed from writing low-level DOM manipulation tasks.
```
OK, "code" is much better.

rockument69 commented on aio/content/guide/architecture.md:
```diff
@@ -1,75 +1,107 @@
 
-  See the [Angular Glossary](guide/glossary) for basic definitions of important Angular terms and usage.
+Modules, components and services are classes that use *decorators*.
+These decorators mark their type and provide metadata that tells Angular how to use them.
```
I think this might work: "Decorators identify the type of class and provide metadata..."

rockument69 commented on aio/content/guide/architecture.md:
```diff
@@ -1,75 +1,107 @@
 
-</div>
+* The metadata for a component class associates it with a [*template*](#templates) that defines a view.
+A template combines ordinary HTML with Angular *directives* and *binding markup* that allow Angular to modify the HTML before rendering it for display.
```
OK, done in next commit.

rockument69 committed:
fixup! docs: Refactor-homepage-architecture
